pygmt/clib/session.py

```diff
@@ -1772,7 +1772,6 @@ def virtualfile_in(
         x=None,
         y=None,
         z=None,
-        extra_arrays=None,
         required_z=False,
         required_data=True,
```

Comment:
> should raise warnings when extra_arrays is used and keep backward compatibility for a few versions (maybe 4 versions).

If we're going to break compatibility of `virtualfile_in` for 4 minor versions, maybe it's a good time to rethink the parameter ordering. Would it make sense for the signature to be something like:

```
def virtualfile_in(
    self,
    check_kind=None,
    required_data=True,
    required_z=False,
    data=None,
    x=None,
    y=None,
    z=None,
    **extra_arrays
```
? We could also mark some of those parameters as keyword only (https://peps.python.org/pep-0570/#keyword-only-arguments) to ensure future compatibility (i.e. prevent positional-only parameters/arguments in case we decide to change the order again).

Reply:
Please see #3836 for my thoughts on the `Session.virtualfile_in` method.